### ISSUE TITLE
Results not reproducible with a LM on wsj0_mix

### ISSUE THREAD

**Issue by vyouman** [espnet/espnet#654]
I used the default parameters of the script without training a language model because I don't have wsj1 transcriptions. I set the language model weight to be 0 and use spa. I used 1 gpu throughout the training and decoding steps with a batchsize of 10. But the result is far from the RESULTS given in this example. Does the language model matter that much in the wsj0_mix task? Do you have any idea?

> CER 
> exp/tr_pytorch_vggblstmp_sde1_rece2_subsample1_2_2_1_1_unit1024_proj1024_d1_unit300_location_spatrue_aconvc10_aconvf100_mtlalpha0.2_adadelta_sampprob0.0_bs10_mli800_mlo150_seed1_lsmunigram0.05_delta/decode_tt_beam30_emodel.acc.best_p0.0_len0.0-0.0_ctcw0.3_rnnlm0.0_1layer_unit1000_sgd_bs300_word65000/min_perm_result.json
> Total Scores: (#C #S #D #I) 476900 70890 50500 40536
> Error Rate:   27.06
> Total Utts:  3000

>  WER
> exp/tr_pytorch_vggblstmp_sde1_rece2_subsample1_2_2_1_1_unit1024_proj1024_d1_unit300_location_spatrue_aconvc10_aconvf100_mtlalpha0.2_adadelta_sampprob0.0_bs10_mli800_mlo150_seed1_lsmunigram0.05_delta/decode_tt_beam30_emodel.acc.best_p0.0_len0.0-0.0_ctcw0.3_rnnlm0.0_1layer_unit1000_sgd_bs300_word65000/min_perm_result.wrd.json
> Total Scores: (#C #S #D #I) 55517 38177 4919 8110
> Error Rate:   51.93
> Total Utts:  3000
>

**Comment by vyouman** [comment]
Some of the log.

> [
> ...
>     {    
>         "main/loss_ctc": 22.921854307651518,
>         "main/loss_att": 50.45514747619629,
>         "main/acc": 0.9903726985324134,
>         "main/cer": 0.0, 
>         "main/wer": 0.0, 
>         "main/loss": 44.9484896850586,
>         "eps": 1e-14,
>         "epoch": 13,
>         "iteration": 42200,
>         "elapsed_time": 55106.551324157976
>     },   
>     {    
>         "main/loss_ctc": 20.196945884227752,
>         "main/loss_att": 47.42536268234253,
>         "main/acc": 0.9927940253499896,
>         "main/cer": 0.0, 
>         "main/wer": 0.0, 
>         "main/loss": 41.979679908752445,
>         "eps": 1e-14,
>         "epoch": 13,
>         "iteration": 42300,
>         "elapsed_time": 55225.99225850799
>     },   
>     {    
>         "main/loss_ctc": 22.334112989902497,
>         "main/loss_att": 49.48581497192383,
>         "main/acc": 0.9921363678916805,
>         "main/cer": 0.0, 
>         "main/wer": 0.0, 
>         "main/loss": 44.05547540664673,
>         "eps": 1e-14,
>         "epoch": 13,
>         "iteration": 42400,
>         "elapsed_time": 55349.36258760397
>     },   
>     {    
>         "main/loss_ctc": 19.977489601373673,
>         "main/loss_att": 48.63069244384766,
>         "main/acc": 0.992552615632488,
>         "main/cer": 0.0, 
>         "main/wer": 0.0, 
>         "main/loss": 42.90005296707153,
>         "eps": 1e-14,
>         "epoch": 13,
>         "iteration": 42500,
>         "elapsed_time": 55468.07696637302
>     }    
> ]

**Comment by vyouman** [comment]
I observed a great performance drop during the training loop.
![Screenshot from 2019-03-17 17-22-33](https://user-images.githubusercontent.com/9965927/54488268-d4b1c000-48da-11e9-91f9-7d62c76f2138.png)

This is the validation log.
![Screenshot from 2019-03-17 17-35-04](https://user-images.githubusercontent.com/9965927/54488292-16426b00-48db-11e9-99a0-94ebb8dff031.png)

**Comment by simpleoier** [comment]
The performance is not satisfactory without the LM. But from the acc and loss in the log, the model was trained moderately well.

**Comment by vyouman** [comment]
Maybe it's my fault. The example script is not for wsj_mix, so I prepared wsj0_mix by myself. I found that the transcriptions I used to prepare the dictionary was wrong. I simply concatenated the text_spk1 and text_spk2 in the train set to generate the text file. Since there are repeated sentences in text_spk* files, the word counts are not accurate. I guess this leads to the performance degradation. I'll fix this bug and rerun the experiment tomorrow.

**Comment by vyouman** [comment]
Recently, I rerun the experiment from the beginning with the language model pretrained with WSJ1 train_si284 transcription and external corpus used to train a language model provided by wsj1. In the last experiment without LM, I mixed the mixtures by myself using python. Since there is performance gap, I install matlab and use the merl tools to mix the speech. However, the performance is still far from satisfactory with the default parameters on the wsj0_mix dataset. Is there some details I have to notice? I trained the model on a single gpu.

**Comment by vyouman** [comment]
tensorboard log:
![Screenshot from 2019-03-25 10-27-53](https://user-images.githubusercontent.com/9965927/54891565-c9692080-4ee8-11e9-8cce-59eb2115f53b.png)
![Screenshot from 2019-03-25 10-28-14](https://user-images.githubusercontent.com/9965927/54891568-ccfca780-4ee8-11e9-9b5c-725197d391aa.png)

**Comment by vyouman** [comment]
LM log:

> 2019-03-23 15:25:55,961 (lm:425) INFO: #sentences in the test data = 333
> 2019-03-23 15:25:55,961 (lm:426) INFO: #tokens in the test data = 5700
> 2019-03-23 15:25:55,961 (lm:427) INFO: oov rate in the test data = 1.16 %
> 2019-03-23 15:25:56,185 (lm:432) INFO: test perplexity: 78.2722592763331
> # Accounting: time=60433 threads=1
> # Ended (code 0) at Sat Mar 23 15:25:57 HKT 2019, elapsed time 60433 seconds

part of the train log:

> [
> ......
> ,
>     {    
>         "main/loss_ctc": 51.39581697463989,
>         "main/loss_att": 55.16699375152588,
>         "main/acc": 0.9782169644289017,
>         "main/cer": 0.0, 
>         "main/wer": 0.0, 
>         "main/loss": 54.41275953292847,
>         "eps": 1e-14,
>         "epoch": 13,
>         "iteration": 42200,
>         "elapsed_time": 52228.657327222
>     },   
>     {    
>         "main/loss_ctc": 51.96717356681824,
>         "main/loss_att": 53.367761306762695,
>         "main/acc": 0.9738968071692605,
>         "main/cer": 0.0, 
>         "main/wer": 0.0, 
>         "main/loss": 53.08764440536499,
>         "eps": 1e-14,
>         "epoch": 13,
>         "iteration": 42300,
>         "elapsed_time": 52343.36450912501
>     },   
>     {    
>         "main/loss_ctc": 54.61860785484314,
>         "main/loss_att": 55.21459522247314,
>         "main/acc": 0.9791530907261287,
>         "main/cer": 0.0, 
>         "main/wer": 0.0, 
>         "main/loss": 55.095398387908936,
>         "eps": 1e-14,
>         "epoch": 13,
>         "iteration": 42400,
>         "elapsed_time": 52462.07859833099
>     },   
>     {    
>         "main/loss_ctc": 46.16356862545013,
>         "main/loss_att": 53.37470134735108,
>         "main/acc": 0.9818268654498553,
>         "main/cer": 0.0, 
>         "main/wer": 0.0, 
>         "main/loss": 51.932475852966306,
>         "eps": 1e-14,
>         "epoch": 13,
>         "iteration": 42500,
>         "elapsed_time": 52578.00563703
>     }    
> ]
>

**Comment by vyouman** [comment]
The result of decoding.

> wer
> + sed -n 2,4p exp/tr_pytorch_vggblstmp_sde1_rece2_subsample1_2_2_1_1_unit1024_proj1024_d1_unit300_location_spatrue_aconvc10_aconvf100_mtlalpha0.2_adadelta_sampprob0.0_bs10_mli800_mlo150_lsmunigram0.05_delta/decode_tt_beam30_emodel.acc.best_p0.0_len0.0-0.0_ctcw0.3_rnnlm1.0_1layer_unit1000_sgd_bs300_word65000/min_perm_result.wrd.json
> Total Scores: (#C #S #D #I) 59016 33986 5611 6549
> Error Rate:   46.80
> Total Utts:  3000

> cer
> + sed -n 2,4p exp/tr_pytorch_vggblstmp_sde1_rece2_subsample1_2_2_1_1_unit1024_proj1024_d1_unit300_location_spatrue_aconvc10_aconvf100_mtlalpha0.2_adadelta_sampprob0.0_bs10_mli800_mlo150_lsmunigram0.05_delta/decode_tt_beam30_emodel.acc.best_p0.0_len0.0-0.0_ctcw0.3_rnnlm1.0_1layer_unit1000_sgd_bs300_word65000/min_perm_result.json
> Total Scores: (#C #S #D #I) 457671 86373 54246 48922
> Error Rate:   31.68
> Total Utts:  3000
> 

Some worse decoding examples.
![Screenshot from 2019-03-25 10-37-49](https://user-images.githubusercontent.com/9965927/54891922-46e16080-4eea-11e9-9eb0-26720e732da0.png)
![Screenshot from 2019-03-25 10-39-17](https://user-images.githubusercontent.com/9965927/54891924-48ab2400-4eea-11e9-8233-58d1a9faddef.png)
And I wonder what does "*" mean in the decoding? Is it padding or something else?

**Comment by simpleoier** [comment]
The "*" means "Insertion" error!
